nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/utils/AzureEventHubUtils.java

```diff
@@ -0,0 +1,90 @@
+
+    public static final String MANAGED_IDENDITY_POLICY = ConnectionStringBuilder.MANAGED_IDENTITY_AUTHENTICATION;
+
+    public static final PropertyDescriptor POLICY_PRIMARY_KEY = new PropertyDescriptor.Builder()
+        .name("Shared Access Policy Primary Key")
+        .description("The primary key of the shared access policy")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+        .sensitive(true)
+        .required(false)
+        .build();
+
+    public static final PropertyDescriptor USE_MANAGED_IDENTITY = new PropertyDescriptor.Builder()
+        .name("use-managed-identity")
+        .displayName("Use Azure Managed Identity")
+        .description("Choose whether or not to use the managed identity of Azure VM/VMSS")
+        .required(false).defaultValue("false").allowableValues("true", "false")
+        .addValidator(StandardValidators.BOOLEAN_VALIDATOR).build();
+
+    public static List<ValidationResult> customValidate(PropertyDescriptor accessPolicyDescriptor,
+        PropertyDescriptor policyKeyDescriptor,
+        ValidationContext context) {
+        List<ValidationResult> retVal = new ArrayList<>();
+
+        boolean accessPolicyIsSet  = context.getProperty(accessPolicyDescriptor).isSet();
+        boolean policyKeyIsSet     = context.getProperty(policyKeyDescriptor).isSet();
+        boolean useManagedIdentity = context.getProperty(USE_MANAGED_IDENTITY).asBoolean();
+
+        if (useManagedIdentity && (accessPolicyIsSet || policyKeyIsSet) ) {
+            final String msg = String.format(
+                "('%s') and ('%s' with '%s') fields cannot be set at the same time.",
+                USE_MANAGED_IDENTITY.getDisplayName(),
+                accessPolicyDescriptor.getDisplayName(),
+                POLICY_PRIMARY_KEY.getDisplayName()
+            );
+            retVal.add(new ValidationResult.Builder().valid(false).explanation(msg).build());
+        } else if (!useManagedIdentity && (!accessPolicyIsSet || !policyKeyIsSet)) {
+            final String msg = String.format(
+                "either('%s') or (%s with '%s') must be set",
+                USE_MANAGED_IDENTITY.getDisplayName(),
+                accessPolicyDescriptor.getDisplayName(),
+                POLICY_PRIMARY_KEY.getDisplayName()
+            );
+            retVal.add(new ValidationResult.Builder().valid(false).explanation(msg).build());
+        }
+        return retVal;
+    }
+
+    public static String getManagedIdentityConnectionString(final String namespace, final String eventHubName){
+        return new ConnectionStringBuilder().setNamespaceName(namespace).setEventHubName(eventHubName)
+                    .setAuthentication(MANAGED_IDENDITY_POLICY).toString();
+    }
+    public static String getSharedAccessSignatureConnectionString(final String namespace, final String eventHubName, final String sasName, final String sasKey) {
```

Comment:
This could also be used from `PutAzureEventHub` now (be careful with the unused import there).

Reply:
done as you suggested